core/src/com/unciv/logic/BarbarianManager.kt

```diff
@@ -0,0 +1,269 @@
+            return (spawnedUnit != null)
+        }
+
+        // Don't spawn wandering barbs too early
+        if (gameInfo.turns < 10)
+            return false
+
+        // Too many barbarians around already?
+        if (tile.getTilesInDistance(4).count { it.militaryUnit?.civInfo == barbarianCiv } > 2)
+            return false
+
+        val canSpawnBoats = gameInfo.turns > 30
+        val validTiles = tile.neighbors.toList().filterNot {
+            it.isImpassible()
+                    || it.isCityCenter()
+                    || it.getFirstUnit() != null
+                    || (it.isWater && !canSpawnBoats)
+                    || (it.hasUnique("Fresh water") && it.isWater) // No Lakes
+        }
+        if (validTiles.isEmpty()) return false
+        val spawnTile = validTiles.random()
+
+        val spawnedUnit = gameInfo.tileMap.placeUnitNearTile(spawnTile.position, chooseBarbarianUnit(spawnTile.isWater), barbarianCiv)
+        return (spawnedUnit != null)
+    }
+
+    private fun chooseBarbarianUnit(naval: Boolean): String {
+        // if we don't make this into a separate list then the retain() will happen on the Tech keys,
+        // which effectively removes those techs from the game and causes all sorts of problems
+        val allResearchedTechs = gameInfo.ruleSet.technologies.keys.toMutableList()
+        for (civ in gameInfo.civilizations.filter { !it.isBarbarian() && !it.isDefeated() }) {
+            allResearchedTechs.retainAll(civ.tech.techsResearched)
+        }
+        val barbarianCiv = gameInfo.getBarbarianCivilization()
+        barbarianCiv.tech.techsResearched = allResearchedTechs.toHashSet()
+        val unitList = gameInfo.ruleSet.units.values
+            .filter { it.isMilitary() }
+            .filter { it.isBuildable(barbarianCiv) }
+
+        var unit = if (naval)
+            unitList.filter { it.isWaterUnit() }.randomOrNull()
+        else
+            unitList.filter { it.isLandUnit() }.randomOrNull()
+
+        if (unit == null) // Didn't find a unit for preferred domain
+            unit = unitList.random() // Pick another, or crash the game vOv
+
+        return unit.name
+    }
+
+    /** When a barbarian is spawned, seed the counter for next spawn */
+    private fun resetCountdown() {
+        // Base 8-12 turns
+        countdown = 8 + Random().nextInt(5)
+        // Quicker on Raging Barbarians
+        if (gameInfo.gameParameters.ragingBarbarians)
+            countdown /= 2
+        // Higher on low difficulties
+        countdown += gameInfo.ruleSet.difficulties[gameInfo.gameParameters.difficulty]!!.barbarianSpawnDelay
+        // Quicker if this camp has already spawned units
+        countdown -= min(3, spawnedUnits)
+
+        countdown *= when (gameInfo.gameParameters.gameSpeed) {
+            GameSpeed.Quick -> 67
+            GameSpeed.Standard -> 100
+            GameSpeed.Epic -> 150
+            GameSpeed.Marathon -> 400 // sic!
+        }
```

Comment:
I would add the /100 here

Reply:
Doesn't work, it's an Int so it would just round down to 0 for Quick